engine/runtime/src/main/java/org/enso/interpreter/runtime/HostClassLoader.java

```diff
@@ -60,26 +66,33 @@ protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundE
       logger.trace("Class {} found in cache", name);
       return l;
     }
-    if (!isRuntimeModInBootLayer && name.startsWith("org.graalvm")) {
-      return polyglotClassLoader.loadClass(name);
-    }
-    if (name.startsWith("org.slf4j")) {
-      // Delegating to system class loader ensures that log classes are not loaded again
-      // and do not require special setup. In other words, it is using log configuration that
-      // has been setup by the runner that started the process. See #11641.
-      return polyglotClassLoader.loadClass(name);
-    }
-    try {
-      l = findClass(name);
-      if (resolve) {
-        l.getMethods();
+    synchronized (INSTANCE) {
```

Comment:
The future of `HostClassLoader` is to have multiple instances, thus this should probably be
```suggestion
    synchronized (this) {
```
otherwise we have problems with this code when removing the singleton nature of the class loader.